src/payment/models/bank-tx/frick.service.ts

```diff
@@ -0,0 +1,291 @@
+    } catch (e) {
+      console.error('Error during get frick transactions:', e);
+    }
+  }
+
+  async getBalance(): Promise<Account[]> {
+    const { accounts } = await this.getAccounts();
+    return accounts;
+  }
+
+  private async getTransactions(fromDate: Date, toDate: Date = new Date()): Promise<Transactions> {
+    const url = `transactions?fromDate=${Util.isoDate(fromDate)}&toDate=${Util.isoDate(
+      toDate,
+    )}&maxResults=2500&status=BOOKED `;
+    return await this.callApi<Transactions>(url);
+  }
+
+  async getAccounts(): Promise<Accounts> {
+    const url = `accounts`;
+    return await this.callApi<Accounts>(url);
+  }
+
+  // --- PARSING --- //
+  private parseTransaction(tx: Transaction): Partial<BankTx> {
+    return {
+      accountServiceRef: tx.orderId ? tx.orderId.toString() : tx.transactionNr?.toString(),
+      bookingDate: tx.valutaIsExecutionDate ? new Date(tx.valuta) : new Date(tx.bookingDate),
+      valueDate: new Date(tx.valuta),
+      txCount: 1,
+      txId: tx.transactionNr?.toString(),
+      ...this.getExchangeInformation(tx),
+      amount: this.getPositiveAmount(tx.totalAmount),
+      instructedAmount: this.getPositiveAmount(tx.fxTransactionAmount),
+      txAmount: this.getPositiveAmount(tx.amount),
+      chargeAmount: this.getPositiveAmount(tx.fees),
+      currency: tx.currency,
+      instructedCurrency: tx.fxTransactionCurrency,
+      txCurrency: tx.currency,
+      chargeCurrency: tx.fees ? tx.currency : undefined,
+      ...this.getCustomerInformation(tx),
+      remittanceInfo: tx.reference,
+      type: tx.type === TransactionType.INTERNAL ? BankTxType.INTERNAL : null,
+      accountIban: tx.direction == TransactionDirection.OUTGOING ? tx.debitor.iban : tx.creditor.iban,
+      txInfo: JSON.stringify(tx),
+    };
+  }
+
+  private getExchangeInformation(tx: Transaction): {
+    exchangeRate: number;
+    exchangeTargetCurrency: string;
+    exchangeSourceCurrency: string;
+  } {
+    if (!tx.md) return null;
+    return {
+      exchangeRate: tx.md == 'D' ? tx.fxrate : Util.round(1 / tx.fxrate, 2),
+      exchangeTargetCurrency: tx.fxTransactionCurrency,
+      exchangeSourceCurrency: tx.currency,
+    };
+  }
+
+  private getCustomerInformation(tx: Transaction): {
+    name?: string;
+    addressLine1?: string;
+    creditDebitIndicator: BankTxIndicator;
+    iban: string;
+    country: string;
+    city: string;
+    memberId: string;
+    bankName: string;
+    bic: string;
+  } {
+    const account = tx.direction == TransactionDirection.OUTGOING ? tx.creditor : tx.debitor;
+    return {
+      name: account.name,
+      addressLine1: account.address,
+      city: account.city,
+      iban: account.iban,
+      memberId: account.accountNumber,
+      country: account.country,
+      bankName: account.creditInstitution,
+      creditDebitIndicator:
+        tx.direction == TransactionDirection.INCOMING ? BankTxIndicator.CREDIT : BankTxIndicator.DEBIT,
```

Comment:
Should we get this indicator from the amount? CREDIT if positive, DEBIT if negative?